sql/core/src/main/scala/org/apache/spark/sql/parquet/newParquet.scala

```diff
@@ -672,7 +672,11 @@ private[sql] object ParquetRelation2 {
         .getKeyValueMetaData
         .toMap
         .get(RowReadSupport.SPARK_METADATA_KEY)
-        .map(DataType.fromJson(_).asInstanceOf[StructType])
+        .map { serializedSchema =>
+          Try(DataType.fromJson(serializedSchema))
```

Comment:
How about output a log for the deprecation also? Just to remind people to use the `Json` style.